openedx/core/djangoapps/safe_sessions/middleware.py

```diff
@@ -412,30 +418,52 @@ def _verify_user(request, response, userid_in_session):
                 # If a view overrode the request.user with a masqueraded user, this will
                 #   revert/clean-up that change during response processing.
                 request.user = request.user.real_user
-            # The user at response time is expected to be None when the user
-            # is logging out.  We won't log that.
-            if request.safe_cookie_verified_user_id != request.user.id and request.user.id is not None:
-                log.warning(
-                    (
-                        "SafeCookieData user at request '{}' does not match user at response: '{}' "
-                        "for request path '{}'"
-                    ).format(  # pylint: disable=logging-format-interpolation
-                        request.safe_cookie_verified_user_id, request.user.id, request.path,
-                    ),
-                )
-                set_custom_attribute("safe_sessions.user_mismatch", "request-response-mismatch")
-            # The user session at response time is expected to be None when the user
-            # is logging out.  We won't log that.
-            if request.safe_cookie_verified_user_id != userid_in_session and userid_in_session is not None:
-                log.warning(
-                    (
-                        "SafeCookieData user at request '{}' does not match user in session: '{}' "
-                        "for request path '{}'"
-                    ).format(  # pylint: disable=logging-format-interpolation
-                        request.safe_cookie_verified_user_id, userid_in_session, request.path,
-                    ),
-                )
-                set_custom_attribute("safe_sessions.user_mismatch", "request-session-mismatch")
+            response_time_user_mismatch = request.safe_cookie_verified_user_id != request.user.id
```

Comment:
"Response time" actually applies to both of these. What do you think of calling this something like `request_object_user_mismatch`? It may be better to add a comment about the verification coming at response-time, rather than trying to get it into the variable names, unless you want `response_time_request_user_mismatch` and `response_time_session_user_mismatch`?